docs/operators/fsInputOperator.md

```diff
@@ -0,0 +1,159 @@
+File Input Operator
+=============
+
+## Operator Objective
+This operator scans a directory for files. Files are then read and split into tuples based on your logic e.g. tuple can be a line  or a block from file, which are then emitted. The operator can be used with different file systems like HDFS, S3, ftp, nfs etc provided hadoop supports that file system.
+
+The default [DirectoryScanner](#https://datatorrent.com/docs/apidocs/com/datatorrent/lib/io/fs/AbstractFileInputOperator.DirectoryScanner.html) polls a input directory for files. You can provide regular expression to scanner to select perticular files from directory. The scanner maintains a list of alredy scanned files, so only newly added files will be read on next poll of directory. The modified or deleted files will not be considered during scan.
+
+The operator is fault tolerant. It tracks previously read files and current offset as part of checkpoint state. In case of failure the operator will skip files that were already processed and fast forward to the offset of the current file. Supports both static and dynamic partitioning. For static partitioning it also accpets changes to number of partitions at run time. The directory scanner is responsible to only accept the files that belong to a partition.
+## Partitioning of File Input Operator
+#### Static Partitioning of File Input Operator
+Configure parameter "partitionCount" to define desired number of static partitions.
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.prop.partitionCount</name>
+    <value>4</value>
+  </property>
+```
+
+where {OperatorName} is the name of the input operator.
+Above lines will partition operator statically 4 times. Above value can be changed accordingly to change the number of static partitions.
+
+#### Dynamic Partitioning of File Input Operator
+Override ```getNewPartitionCount()``` method to do dymanic partitioning.
```

Comment:
The logic inside getNewPartitionCount should be such that value will be calculated based on load or other factors to achieve dynamic partitioning.
In other case "partitionCount" can be changed at runtime, but that would still be static partitioning as it doesn't undergo re-partition based on some logic as such.
Isn't this understanding right?